parserator_web/templates/parserator_web/index.html

```diff
@@ -39,7 +39,62 @@ <h4>Parsing results</h4>
 
 {% block extra_js %}
   <script type="text/javascript">
-    /* TODO: Flesh this out to connect the form to the API and render results
-    in the #address-results div. */
+
+    function createRow(arr) {
+      return `<tr><td>${arr[1]}</td><td>${arr[0]}</td></tr>`;
+    }
+
+    function renderAddress(data) {
+      const addressDiv = document.getElementById('address-results');
+      addressDiv.style.display = 'block';
+      const address_components = Object.entries(data.address_components);
+      addressDiv.innerHTML = '';
+      addressDiv.innerHTML = `
+      <h4>Parsing results</h4>
+        <p>Address type: ${data.address_type} <strong><span id="parse-type"></span></strong></p>
+        <table class="table table-bordered">
+          <thead>
+            <tr>
+            <th>Address part</th>
+            <th>Tag</th>
+          </tr>
+          </thead>
+          <tbody>
+            ${address_components.map(e => createRow(e)).join('')}
+          </tbody>
+        </table>`;
+    }
+
+    function renderNullError() {
+      const addressDiv = document.getElementById('address-results');
+      addressDiv.style.display = 'block';
+      addressDiv.innerHTML = '';
+      addressDiv.innerHTML = '<div class="alert alert-danger">Please provide a value to parse.</div>';
+    }
+
+    function renderError(error) {
+      const addressDiv = document.getElementById('address-results');
+      addressDiv.style.display = 'block';
+      addressDiv.innerHTML = '';
+      addressDiv.innerHTML = `<div class="alert alert-danger">${error}</div>`;
+    }
+
+    const searchForm = document.querySelector('.form')
+    searchForm.addEventListener('submit', e => {
+      e.preventDefault();
+      const addressForm = document.getElementById('address');
+      const inputString = addressForm.value;
+      addressForm.value = '';
+      const url = new URL('http://localhost:8000/api/parse/');
```

Comment:
How about omitting the host from this URL, that way this would work whether we're running the app locally or in a deployment environment? 

```suggestion
      const url = new URL('/api/parse/');
```

Reply:
Thank you for this feedback! Completely agree that a hardcoded URL is not ideal. I've updated the URL constructor so it uses a 'base URL' variable that can be changed when deploying to production, and the relative URL path remains the same.